### PR TITLE
Lazy loading

### PR DESCRIPTION
**Pull request by etokareva**

### CODE REVIEW

File: themes/qgrid/layout/layout.ejs [hunk 1]
```diff
@@ -51,6 +51,7 @@
 
     <!-- this needs to be loaded before guide's inline scripts -->
     <script src="<%- url_for('/js/menu.js') %>"></script>
+    <script src="<%- url_for('/js/lazy-loading.js') %>"></script>
```
Comment (klumba12) [inline themes/qgrid/layout/layout.ejs]: call it just lazy.js

File: themes/qgrid/source/js/lazy-loading.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+function isVisible(elem) {
```
Comment (klumba12) [inline themes/qgrid/source/js/lazy-loading.js]: just a few names:
coords -> rect
elem -> element

File: themes/qgrid/source/js/lazy-loading.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+	const windowHeight = document.documentElement.clientHeight;
+
+	const topVisible = coords.top > 0 && coords.top < windowHeight;
+	const bottomVisible = coords.bottom < windowHeight && coords.bottom > 0;
+
+	return topVisible || bottomVisible;
+}
+
+function showVisible() {
+	const editors = document.querySelectorAll('.editor iframe');
```
Comment (klumba12) [inline themes/qgrid/source/js/lazy-loading.js]: use `data-src` as selector?

File: themes/qgrid/source/js/lazy-loading.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+	const windowHeight = document.documentElement.clientHeight;
+
+	const topVisible = coords.top > 0 && coords.top < windowHeight;
+	const bottomVisible = coords.bottom < windowHeight && coords.bottom > 0;
+
+	return topVisible || bottomVisible;
+}
+
+function showVisible() {
```
Comment (klumba12) [inline themes/qgrid/source/js/lazy-loading.js]: rename to `liveUp`

File: themes/qgrid/source/js/lazy-loading.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
+function isVisible(elem) {
+
+	const coords = elem.getBoundingClientRect();
+
+	const windowHeight = document.documentElement.clientHeight;
+
+	const topVisible = coords.top > 0 && coords.top < windowHeight;
+	const bottomVisible = coords.bottom < windowHeight && coords.bottom > 0;
+
+	return topVisible || bottomVisible;
```
Comment (klumba12) [inline themes/qgrid/source/js/lazy-loading.js]: will it work correctly in all cases? even if example is more then windowHeight?

File: themes/qgrid/source/js/lazy-loading.js (hunk 3 above)
Comment (etokareva) [reply themes/qgrid/source/js/lazy-loading.js]: yes) I just thought about it

**Review by klumba12 — APPROVED** [review]

**Pull request merged**